app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -83,11 +83,16 @@ class FormController(
                    cache.form.thirdPartyData,
                    cache.formTemplate)
                    .map {
-                     case (validationResult, v, _) =>
-                       (
-                         ValidationUtil.isFormValid(validationResult.toMap),
-                         FormData(validationResult.flatMap(_._2.toFormField)),
-                         v)
+                     case (validationResult, validatedType, _) => {
```

Comment:
Extract the body of this case into a private function. It's too big to be inline.